### PR TITLE
Add GCF function to create a GCS object.

### PR DESCRIPTION
**Pull request by jiggoha**
This function will be called by the Cloud Build trigger defined at [trillian-examples/](https://github.com/google/trillian-examples/)cloudbuild_docker.yaml.

### CODE REVIEW

File: serverless/experimental/gcp-log/gcs_uploader/function.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
```
Comment (AlCutter) [inline serverless/experimental/gcp-log/gcs_uploader/function.go:1]: 2022? :)

File: serverless/experimental/gcp-log/gcs_uploader/function.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	gcs "cloud.google.com/go/storage"
+)
+
+// CreateGCSObject is writes a GCS object with EntryContent to EntryPath in
```
Comment (AlCutter) [inline serverless/experimental/gcp-log/gcs_uploader/function.go:27]: nit: "is writes"

also, should `EntryContent` etc. be `entryContent` to match the JSON naming?

File: serverless/experimental/gcp-log/gcs_uploader/function.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	client, err := gcs.NewClient(ctx)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to create GCS client: %q", err),
+			http.StatusInternalServerError)
+		return
+	}
+
+	obj := client.Bucket(d.Bucket).Object(d.EntryPath)
+	writer := obj.NewWriter(ctx)
+	if _, err := fmt.Fprintf(writer, d.EntryContent); err != nil {
```
Comment (AlCutter) [inline serverless/experimental/gcp-log/gcs_uploader/function.go:67]: Could this use the `Writer` interface directly with something along the lines of this:

```go
if n, err := writer.Write([]byte(d.EntryContent)); err != nil || n != len(d.EntryContent) {
  ...
}
```
?

File: serverless/experimental/gcp-log/gcs_uploader/go.sum [hunk 4]
```diff
@@ -0,0 +1,595 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (AlCutter) [inline serverless/experimental/gcp-log/gcs_uploader/go.sum:1]: I wonder if a `go mod tidy` would help cut the number of versions in here down?

File: serverless/experimental/gcp-log/gcs_uploader/function.go (hunk 1 above)
Comment (jiggoha) [reply serverless/experimental/gcp-log/gcs_uploader/function.go]: Time flies!

File: serverless/experimental/gcp-log/gcs_uploader/function.go (hunk 3 above)
Comment (jiggoha) [reply serverless/experimental/gcp-log/gcs_uploader/function.go]: Neat!

File: serverless/experimental/gcp-log/gcs_uploader/go.sum (hunk 4 above)
Comment (jiggoha) [reply serverless/experimental/gcp-log/gcs_uploader/go.sum:1]: Done.

**Review by AlCutter — APPROVED** [review]

**Pull request merged**